Doc/library/sqlite3.rst

```diff
@@ -47,85 +47,179 @@ This document includes four main sections:
 
 Tutorial
 --------
 
-To use the module, start by creating a :class:`Connection` object that
-represents the database.  Here the data will be stored in the
-:file:`example.db` file::
+In this tutorial, you will create a database of Monty Python movies
+using basic :mod:`!sqlite3` functionality.
+It assumes a fundamental understanding of database concepts,
+including `cursors`_ and `transactions`_.
+
+First, we need to create a new database and open
+a database connection to allow :mod:`!sqlite3` to work with it.
+Call :func:`sqlite3.connect` to to create a connection to
+the database :file:`tutorial.db` in the current working directory,
+implicitly creating it if it does not exist::
 
    import sqlite3
-   con = sqlite3.connect('example.db')
+   con = sqlite3.connect("tutorial.db")
 
-The special path name ``:memory:`` can be provided to create a temporary
-database in RAM.
+The returned :class:`Connection` object ``con``
+represents the connection to the on-disk database.
 
-Once a :class:`Connection` has been established, create a :class:`Cursor` object
-and call its :meth:`~Cursor.execute` method to perform SQL commands::
+In order to execute SQL statements and fetch results from SQL queries,
+we will need to use a database cursor.
+Call :meth:`con.cursor() <Connection.cursor>` to create the :class:`Cursor`::
 
    cur = con.cursor()
 
-   # Create table
-   cur.execute('''CREATE TABLE stocks
-                  (date text, trans text, symbol text, qty real, price real)''')
+Now that we've got a database connection and a cursor,
+we can create a database table ``movie`` with columns for title,
+release year, and review score.
+For simplicity, we can just use column names in the table declaration --
+thanks to the `flexible typing`_ feature of SQLite,
+specifying the data types is optional.
+Execute the ``CREATE TABLE`` statement
+by calling :meth:`cur.execute(...) <Cursor.execute>`::
+
+   cur.execute("CREATE TABLE movie(title, year, score)")
+
+.. Ideally, we'd use sqlite_schema instead of sqlite_master below,
+   but SQLite versions older than 3.33.0 do not recognise that variant.
+
+We can verify that the new table has been created by querying
+the ``sqlite_master`` table built-in to SQLite,
+which should now contain an entry for the ``movie`` table definition
+(see `The Schema Table`_ for details).
+Execute that query by calling :meth:`cur.execute(...) <Cursor.execute>`,
+assign the result to ``res``,
+and call :meth:`res.fetchone() <Cursor.fetchone>` to fetch the resulting row::
+
+   >>> res = cur.execute("SELECT name FROM sqlite_master")
+   >>> res.fetchone()
+   ('movie',)
+
+As expected, the query shows the table has been created,
+as it returns a :class:`tuple` containing the table's name.
+As an exercise, try querying ``sqlite_master``
+for a non-existent table ``abc``::
+
+   >>> res = cur.execute("SELECT name FROM sqlite_master WHERE name='abc'")
+   >>> res.fetchone()
+   >>>
+
+Indeed, the query returns an empty result.
```

Comment:
Usually exercises are left to the reader, but here we are immediately showing what happens like we did throughout the rest of the tutorial, so I'm not sure why it's called "exercise".  I would remove that and simplify a bit.
In addition, I changed the name of the non-existing table to a more fitting `spam`.
```suggestion
If we try querying ``sqlite_master`` for a non-existent table ``'spam'``,
:meth:`!res.fetchone()` will return ``None``::

   >>> res = cur.execute("SELECT name FROM sqlite_master WHERE name='spam'")
   >>> res.fetchone()
   >>>

Indeed, the query returns an empty result.
```

Reply:
I take it you haven't actually pushed the changes yet?

Reply:
Actually I meant to remove the "Indeed, the query returns an empty result." after the example, since it's now mentioned before it.  I now updated the suggestion above.

Reply:
> I take it you haven't actually pushed the changes yet?

I pushed them, but this change was never applied :)